Userland/Utilities/strace.cpp

```diff
@@ -36,24 +137,328 @@ static void handle_sigint(int)
+    void add_argument(T&& arg)
+    {
+        add_argument("{}", forward<T>(arg));
+    }
+
+    void add_string_argument(Syscall::StringArgument const& string_argument)
+    {
+        if (string_argument.characters == nullptr)
+            add_argument("null");
+        else {
+            auto string = copy_from_process(string_argument.characters, string_argument.length);
+            add_argument("\"{}\"", StringView(string.data(), string.size()));
+        }
+    }
+
+    template<typename... Ts>
+    void add_arguments(Ts&&... args)
+    {
+        (add_argument(forward<Ts>(args)), ...);
+    }
+
+    template<typename T>
+    void format_result_no_error(T res)
+    {
+        m_builder.appendff(") = {}\n", res);
+    }
+
+    void format_result(Integral auto res)
+    {
+        m_builder.append(") = ");
+        if (res < 0)
+            m_builder.appendff("{} {}", res, errno_name(-(int)res));
+        else
+            m_builder.appendff("{}", res);
+        m_builder.append('\n');
+    }
+
+    void format_result(void* res)
+    {
+        m_builder.appendff(") = {}\n", res);
+    }
+
+    StringView string_view()
+    {
+        return m_builder.string_view();
+    }
+
+private:
+    void add_argument_separator()
+    {
+        if (!m_first_arg) {
+            m_builder.append(", ");
+        }
+        m_first_arg = false;
+    }
+
+    StringBuilder m_builder;
+    bool m_first_arg { true };
+};
+
+static void format_open(FormattedSyscallBuilder& builder, Syscall::SC_open_params* params_p)
+{
+    auto params = copy_from_process(params_p);
+
+    if (params.dirfd == AT_FDCWD)
+        builder.add_argument("AT_FDCWD");
+    else
+        builder.add_argument(params.dirfd);
+
+    builder.add_string_argument(params.path);
+
+    Vector<StringView> active_flags;
+    if (params.options & O_RDWR)
+        active_flags.append("O_RDWR");
+    else if (params.options & O_RDONLY)
+        active_flags.append("O_RDONLY");
+    else if (params.options & O_WRONLY)
+        active_flags.append("O_WRONLY");
+
+    if (params.options & O_APPEND)
+        active_flags.append("O_APPEND");
+    if (params.options & O_CREAT)
+        active_flags.append("O_CREAT");
+    // TODO: etc...
+
+    // TODO: add to FormattedSyscallBuilder
+    StringBuilder sbuilder;
+    sbuilder.join(" | ", active_flags);
+    builder.add_argument(sbuilder.to_string());
+
+    if (params.options & O_CREAT)
+        builder.add_argument("{:04o}", params.mode);
+}
+
+namespace AK {
+template<>
+struct Formatter<struct timespec> : StandardFormatter {
+    Formatter() = default;
+    explicit Formatter(StandardFormatter formatter)
+        : StandardFormatter(formatter)
+    {
+    }
+
+    void format(FormatBuilder& format_builder, struct timespec value)
+    {
+        auto& builder = format_builder.builder();
+        builder.append('{');
+        builder.appendff("tv_sec={}, tv_nsec={}", value.tv_sec, value.tv_nsec);
+        builder.append('}');
```

Comment:
```suggestion
        builder.appendff("{{ tv_sec={}, tv_nsec={} }}", value.tv_sec, value.tv_nsec);
```

Reply:
Nice! I suspected there was a escaping method, but I tried with backslash and didn't work. It's obvious now in hindsight.

Reply:
FIxed now.